Controllers/UserController.js

```diff
@@ -0,0 +1,64 @@
+import path from 'path'
+import { homedir } from 'os'
+import * as fs from 'fs'
+import * as dotenv from 'dotenv'
+import { fileURLToPath } from 'url'
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = path.dirname(__filename)
+dotenv.config({path: path.join(__dirname, '../.env')})
+import * as readline from 'node:readline/promises';
+import { stdin as input, stdout as output } from 'node:process';
+import { createClient } from '@supabase/supabase-js'
+
+const supabaseUrl = process.env.SUPABASE_URL
+const supabaseAnonKey = process.env.SUPABASE_ANON_KEY
+const supabase = createClient(supabaseUrl, supabaseAnonKey)
+
+const userHomeDir = homedir()
+
+const clearLastLine = () => {
+  process.stdout.moveCursor(0, -1) // up one line
+  process.stdout.clearLine(1) // from cursor to end
+}
+
+
+export const register = async (email, password) => {
+    const {data, error} = await supabase.auth.signUp({
+      email,
+      password,
+    })
+    if(error){
+      console.log(error.message)
+    }else{
+      console.log('User created successfully')
+      // save user data and session locally i think data.user??
+    }
+}
+
+
+export const login = async (email, password, channel) => {
+    const { data, error } = await supabase.auth.signInWithPassword({
+      email,
+      password,
+    })
+    if(error){
+      console.log(error.message)
+    }else{
+      console.log("Login Succcesfull")
+      //console.log(data);
+      const userData = data
+      // save user/session data && current channel to file
+      fs.writeFileSync(userHomeDir + '/.boxrc.json', JSON.stringify({channel, userData})); 
```

Comment:
My PR only saves the session data e.i. `data.session` to `.boxrc.json`. What else is in the `data` object other than session data?